### ISSUE TITLE
Tasks with target = TaskScope.Global as the scope do not show up in task picker UI

### ISSUE THREAD

**Issue by g-arjones** [Microsoft/vscode#40515]
- VSCode Version: Code 1.18.1 (929bacba01ef658b873545e26034d1a8067445e9, 2017-11-16T18:23:26.125Z)
- OS Version: Darwin x64 17.3.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
code-runner|for|0.8.4
cpptools|ms-|0.14.2
Ruby|reb|0.15.0

---

Steps to Reproduce:

1. Create a vscode.Task using the new API passing vscode.TaskScope.Global as the target

Reproduces without extensions: Yes

I am developing an extension that provides tasks for each folder in a multiroot project and a few tasks that are applicable to all folders. Maybe I am missing something but the tasks that have TaskScope.Globall as the target do not show up in the task picker UI. Is this behavior intentional?

**Comment by dbaeumer** [comment]
This is clearly a bug.

**Comment by g-arjones** [comment]
Bummer.. Where you able to reproduce?

**Comment by dbaeumer** [comment]
I haven't looked into tracking down what causes this. But if your description is true then it is clearly a bug.

**Comment by dbaeumer** [comment]
I was able to reproduce this but need to think about it a little more. The problem is how to resolve some variable like `${workspaceFolder}` in the context of a global task. We should simply replace them the variable name to make clear what happens.

**Comment by g-arjones** [comment]
Yeah, I had a look into this myself and did not find any codepath that would make this work. Seems more like a lack of implementation than a bug. The task service skips definitions that are not associated with a workspaceFolder.

I am not sure I understand your point about the variable resolution though, if the task is global ${workspaceFolder} just doesn't exist in such context and resolving it to an empty string/undefined would make sense, wouldn't it?

**Comment by g-arjones** [comment]
As far as I can understand, [this is what is causing global tasks to be skipped](https://github.com/Microsoft/vscode/blob/a36ef3d7888cae91621193e0bf1be0d2f69f2688/src/vs/workbench/parts/tasks/electron-browser/task.contribution.ts#L1304)

**Comment by dbaeumer** [comment]
Yes, minus that we decided to keep the `${workspaceFolder}` as a string

**Comment by dbaeumer** [comment]
The right fix would be:

- have global task separate since they can't be modified using a tasks.json and we shouldn't try to do so.
- variables that depend on a workspace folder should keep their value.

Leaving as a bug.

**Comment by g-arjones** [comment]
Do you have a plan/overview of what needs to be done? I could give this a try...

**Comment by dbaeumer** [comment]
Not yet. It is not the best issue to start with a PR for tasks. I am not sure yet if we need to separate tasks into two buckets or if we filter them for certain actions. For example Tasks > Configure Task can not show a global task.

**Comment by g-arjones** [comment]
@dbaeumer any plans of fixing this? it's been almost an year... thanks!

**Comment by g-arjones** [comment]
@dbaeumer It's been 7 years... Any chance this will be fixed anytime soon?
